cli/cook/subcommands/wait.py

```diff
@@ -0,0 +1,61 @@
+import time
+
+from cook.subcommands.show import query, print_no_data, seconds_to_timedelta
+
+from cook.util import strip_all, print_info
+
+
+def all_jobs_completed(jobs):
+    """Returns jobs if they are all completed, otherwise False."""
+    if all(j.get('status') == 'completed' for j in jobs):
+        return jobs
+    else:
+        return False
+
+
+def all_instances_completed(instances):
+    """Returns instances if they are all completed, otherwise False."""
+    if all(i.get('status') == 'completed' for i in instances):
+        return instances
+    else:
+        return False
+
+
+def all_groups_completed(groups):
+    """Returns groups if they are all completed, otherwise False."""
+    if all(len(g.get('jobs')) == g.get('completed') for g in groups):
+        return groups
+    else:
+        return False
+
+
+def wait(clusters, args):
+    """Waits for jobs / instances / groups with the given UUIDs to complete."""
+    timeout = args.get('timeout')
```

Comment:
If not passed, do we ultimately only end up waiting for up to 30 seconds?

Reply:
No, we will wait indefinitely by default

Reply:
OK.  I must have misunderstood the code.

Reply:
Here's an example with the default (indefinite) timeout:

```bash
$ time cs wait $(cs --silent submit sleep 45)
Will wait indefinitely.
Waiting for the following jobs on dev0: 709c8bcb-a3f5-4643-a50d-00bcc3e95dc4 ... Done

real    0m51.092s
user    0m0.852s
sys     0m0.056s

```